sql/sql_connect.cc

```diff
@@ -655,6 +649,49 @@ void update_global_user_stats(THD* thd, bool create_user, time_t now)
   mysql_mutex_unlock(&LOCK_global_user_client_stats);
 }
 
+static void clear_stats_concurrent_connections(HASH* stats) {
+  for (ulong idx=0; idx < stats->records; idx++)
+  {
+    USER_STATS* user_stats=
+      reinterpret_cast<USER_STATS*>(my_hash_element(stats, idx));
+    user_stats->concurrent_connections = 0;
+  }
+}
+
+static void inc_stats_concurrent_conn(HASH* stats, const char* user_string, int cnt) {
+  USER_STATS* user_stats;
+  if ((user_stats = (USER_STATS *) my_hash_search(stats,
+                                                  (uchar *) user_string,
+                                                  strlen(user_string))))
+  {
+    user_stats->concurrent_connections += cnt;
+  }
+}
+
+// Updates number of concurrent connections for user_stats and client_stats
+// based on account resoure limits
```

Comment:
If a header comment is added (strictly speaking it has to be added for all the new functions, but the previous two are trivial enough so that's not a critical issue), then it has to be a Doxygen one